feat: invalidate message and fix HMR for HOC, class component & styled components (#79)
vitejs/vite-plugin-react: packages/plugin-react/README.md

```
Uncaught Error: @vitejs/plugin-react can't detect preamble. Something is wrong.
```

## Consistent components exports

> For React refresh to work correctly, your file should only export React components. The best explanation I've read is the one from the [Gatsby docs](https://www.gatsbyjs.com/docs/reference/local-development/fast-refresh/#how-it-works).

Reviewer: ```suggestion
For React refresh to work correctly, your file should only export React components. You can find a good explanation in the [Gatsby docs](https://www.gatsbyjs.com/docs/reference/local-development/fast-refresh/#how-it-works).
```

Author: I will also update the wording for the SWC plugin
